torch/distributed/_tensor/device_mesh.py

```diff
@@ -283,6 +283,8 @@ def scatter(
         Returns:
             A :class:`Work` object
         """
+        if output.is_meta:
```

Comment:
Oh i see you added the check here to avoid communication. 

ideally we should use the meta tensor way (to register a meta kernel for the collective op) so that it would avoid the communication. But I think the collective meta registration part still being worked on by @wconstab so we can probably do that later.

Do you also need to avoid broadcast?

Reply:
Yes, we do. Let me add it.